### PR TITLE
Fix TypedDict inherit-non-class false-positive Python 3.9+

### LINKED ISSUE #1927: `mypy_extensions.TypedDict` errors because when creating a class, it is inferred as an instance
### Steps to reproduce
1. Create a new variable `Dict1 = mypy_extensions.TypedDict('Dict1', {'a' : int})`.
2. Create a class `class Dict2(Dict1, Total=True): pass`
3. Run pyling

### Current behavior
Pylint gives an error for the class creation as inheriting from a non class is invalid.

### Expected behavior
No error as this valid.

### pylint --version output
Python 3.6.4rc1

### PR DESCRIPTION
**Pull request by cdce8p**
## Steps

- [ ] Add yourself to CONTRIBUTORS if you are a new contributor.
- [x] Add a ChangeLog entry describing what your PR does.
- [ ] If it's a new feature or an important bug fix, add a What's New entry in `doc/whatsnew/<current release.rst>`.
- [x] Write a good description on what the PR does.

## Description
In Python 3.9 `typing.TypedDict` changed from a class to function with some metaclass magic.
This caused a `inherit-non-class` error for valid code.

**Source code**
[Python 3.8 - typing.py](https://github.com/python/cpython/blob/c370596a23167749b8127d1d2e20e039865aa695/Lib/typing.py#L1760-L1763)
[Python 3.9 - typing.py](https://github.com/python/cpython/blob/48d16b4b0fb0d0558bdbc1aee850650a671ecc77/Lib/typing.py#L1940-L1998)

**Documentation**
https://docs.python.org/3/library/typing.html#typing.TypedDict

## Type of Changes

|   | Type |
| ------------- | ------------- |
| ✓  | :bug: Bug fix  |
|   | :sparkles: New feature |
|   | :hammer: Refactoring  |
|   | :scroll: Docs |

## Related Issue

Closes #1927

### CODE REVIEW

**Review by Pierre-Sassoulas — APPROVED** [review]

**Comment by Pierre-Sassoulas** [comment]
The fix seems clean enough even if it's not generic aka "make astroid understand the metaclass magic".  Will merge when the master branch is fixed.

**Comment by cdce8p** [comment]
I just opened a MR in astroid to fix it there: https://github.com/PyCQA/astroid/pull/915
Once that is merged, this one could be reverted.

**Comment by Pierre-Sassoulas** [comment]
Well, except the tests and the changelog, I like those ;)

**Pull request merged**
